pkg/cloud/services/compute/instance.go

```diff
@@ -610,16 +609,29 @@ func (s *Service) getImageID(imageName string) (string, error) {
 	}
 }
 
-func (s *Service) AssociateFloatingIP(instanceID, floatingIP string) error {
-	opts := floatingips.AssociateOpts{
-		FloatingIP: floatingIP,
+func (s *Service) GetControlPlanePort(instance *infrav1.Instance) (*ports.Port, error) {
```

Comment:
GetPort is better because this is used for getting port of bastion instance also.

Reply:
I'm at risk of bike-shedding myself here :)

I want to differentiate this port from other ports because it's 'special'. It's the port that will have an IP on NodeCIDR, and it's the only port which we can guarantee is on the correct subnet to associate a floating ip with. It's the port whose first fixed IP will populate instance.IP.

What kind of special? I called it the 'primary' port first, but I didn't like that and changed it to the 'control plane' port. That's an overloaded term, though, and potentially confusing when applied to the bastion host as you point out. I guess it could be the 'management' port?

It's not just any port, though: it's this specific port. It's the <insert name here> port 😂

I don't want to get too caught up on this, though, because I think I'm going to have to come back to this stuff soon anyway. I need to address the issue of `GetIPFromInstance` returning only the first internal port without sorting, so I'm fully expecting to rework this anyway. For example, ideally I think we would determine both this port and instance.ID from cluster.Subnet.

I'm going to go with 'management' port for now, and I'll add an approriate comment block explaining what differentiates it from other ports. If that doesn't work I'll go to GetPort and come back to the issue of what to call it while tackling the wider issue. I'm guessing I should probably also open an issue to discuss the other thing first?